plugins/modules/credential.py

```diff
@@ -0,0 +1,141 @@
+      type: str
+    description:
+      description:
+        - The description of the credential.
+      required: False
+      type: str
+    username:
+      description:
+        - The username used to access the project repository.
+      required: True
+      type: str
+    secret:
+      description:
+        - The secret associated with the credential (either a password or access token).
+      required: True
+      type: str
+    credential_type:
+      description:
+        - The type of the credential.
+      choices: ["GitHub Personal Access Token", "GitLab Personal Access Token", "Container Registry"]
+      default: "GitHub Personal Access Token"
+      type: str
+    state:
+      description:
+        - Desired state of the resource.
+      choices: ["present", "absent"]
+      default: "present"
+      type: str
+
+extends_documentation_fragment: infra.eda_configuration.auth
+"""
+
+
+EXAMPLES = """
+- name: Create eda credential
+  infra.eda_configuration.credential:
+    name: my_credential
+    description: my github access credential
+    username: derekwaters
+    secret: this_is_not_a_real_token
+    credential_type: "GitHub Personal Access Token"
+    state: present
+    eda_host: eda.example.com
+    eda_username: admin
+    eda_password: Sup3r53cr3t
+
+"""
+
+from ..module_utils.eda_module import EDAModule
+
+
+def main():
+    # Any additional arguments that are not fields of the item can be added here
+    argument_spec = dict(
+        name=dict(required=True),
+        new_name=dict(),
+        description=dict(),
+        username=dict(required=True),
+        secret=dict(required=True),
+        credential_type=dict(choices=["GitHub Personal Access Token",
+                                      "GitLab Personal Access Token",
+                                      "Container Registry"],
+                             default="GitHub Personal Access Token"),
+        state=dict(choices=["present", "absent"], default="present"),
+    )
+
+    # Create a module for ourselves
+    module = EDAModule(argument_spec=argument_spec)
+
+    # Extract our parameters
+    name = module.params.get("name")
+    new_name = module.params.get("new_name")
+    state = module.params.get("state")
+
+    new_fields = {}
+
+    # Attempt to look up an existing item based on the provided data
+    existing_item = module.get_one("credentials", name_or_id=name, key="req_url")
+
+    if state == "absent":
+        # If the state was absent we can let the module delete it if needed, the module will handle exiting from this
+        module.delete_if_needed(existing_item, key="req_url")
+
+    # Create the data that gets sent for create and update
+    # Remove these two comments for final
+    # Check that Links and groups works with this.
+    new_fields["name"] = new_name if new_name else (module.get_item_name(existing_item) if existing_item else name)
+    for field_name in (
+        "description",
+        "credential_type",
+    ):
```

Comment:
You're missing a little bit in here to make the secret and username actually be applied by the module. Upon testing adding the following two lines makes it work. I'm not sure if you would have managed to get something actually created with a username and secret beforehand.

```suggestion
    for field_name in (
        "description",
        "credential_type",
        "secret",
        "username",
    ):
```

Reply:
Yes, my fault, I think I dropped those lines somewhere between moving code between branches. I added these lines, and retested a GitHub credential with token was added, and a project using the credential syncs correctly.